pkg/psmdb/mongos.go

```diff
@@ -64,8 +63,8 @@ func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod,
 		annotations = make(map[string]string)
 	}
 
-	if c := cr.Spec.Sharding.Mongos.Configuration; c != "" && configSource == VolumeSourceConfigMap {
```

Comment:
but this config provided by user directly in cr. We should handle it too

Reply:
yes, it's handled from a config map to simplify the logic

configuration from cr.yaml goes to config map and from there it's handled as any other custom configuration